from google.cloud import bigquery
from google.api_core.retry import Retry
from google.cloud.exceptions import GoogleCloudError
from google.api_core.exceptions import RetryError,DeadlineExceeded
 
import builtins
 
project_id = 'tnt01-odycda-bld-01-1b81'
 
try:
    client = bigquery.Client(project=project_id)
    print("Attempting to connect to Bigquery..")
except Exception as e:
    print(f"Error while initializing the bigquery client: {e}")
 
no_retries = Retry(predicate=lambda exc:False)
 
timeout = 10
 
query = "SELECT * FROM `tnt01-odycda-bld-01-1b81.tnt01_odycda_bqd_euwe2_tdm_control.file_meta_data_info` LIMIT 1000"
 
try:
    query_job = client.query(query,retry=no_retries,timeout=timeout)
    results = query_job.result()
    for row in results:
            print(row)
 
except GoogleCloudError as gce:
     print(f"Google cloud error : {gce}")
 
except RetryError:
     print("Retry Error : Maximum number of  retries exceeded without successful request.")
 
except DeadlineExceeded:
     print("Deadline Exceeded: the request did not complete in the specified time")
 
except builtins.ConnectionError:
     print("Connection Error: Failed to connect Big Query within the timeout.")
 
 
except Exception as e:
     print(f"General Error: {e}")